feat(instances): understanding qga (#3896)
scaleway/docs-content: compute/instances/reference-content/understanding-qemu-guest-agent.mdx

---
meta:
  title: Understanding the QEMU Guest Agent in Scaleway Instances

>   description: Discover how Scaleway Instance pricing works, including hourly rates, billing, and tips to manage costs effectively.

Reviewer: I suspect this needs to be adjusted to match the document.

Author: Indeed, I forgot to update it. Done now.

>   description: Discover how the QEMU Guest Agent in Scaleway Instances works.
> content:
>   h1: Understanding the QEMU Guest Agent in Scaleway Instances
>   paragraph: Discover how the QEMU Guest Agent in Scaleway Instances works.

Reviewer 2: ```suggestion
  description: Discover how the QEMU Guest Agent works with Scaleway Instances.
content:
  h1: Understanding the QEMU Guest Agent in Scaleway Instances
  paragraph: Discover how the QEMU Guest Agent works with Scaleway Instances.
```

>   paragraph: Discover how Scaleway Instance pricing works, including hourly rates, billing, and tips to manage costs effectively.

Reviewer: Here too.

tags: instance qga guemu guest agent
dates:
  validation: 2024-10-28
categories:
  - compute
---

Some features of the Instances product require Scaleway's infrastructure to query or exchange information with your Instance. To enable this communication, a software component must run on the guest operating system: the QEMU Guest Agent (QGA).

> This document provides essential insights into this mechanism.

Reviewer 2: ```suggestion
This page provides essential insights into this mechanism.
```

<Message type="note">

>   This document does not apply to Instances running the Microsoft Windows operating system.

Reviewer 2: ```suggestion
  This documentation page does not apply to Instances running the Microsoft Windows operating system.
```

</Message>

## What are the features provided by QGA?

Running the QEMU Guest Agent (QGA) on your Instance currently enables the following feature:

> - **Automatic network reconfiguration** upon flexible IP attachment or detachment [Learn how to enable/disable this feature]().

Reviewer 2: ```suggestion
- **Automatic network reconfiguration** upon flexible IP attachment or detachment [Learn how to enable/disable this feature]().
```
Missing link here

Additional features may be added in the future.

> ## Checking the status of QGA

Reviewer 2: ```suggestion
## Checking QGA's status
```

> Since March 1, 2024, all Scaleway-provided GNU/Linux and InstantApp images for Instances come with QGA pre-installed and enabled by default.

Reviewer 2: ```suggestion
Since March 1st, 2024, all Scaleway-provided GNU/Linux and InstantApp images for Instances come with QGA pre-installed and enabled by default.
```

To verify that QGA is running on your Instance, use the following command:

```bash
# systemctl is-active qemu-guest-agent.service
```

If the output is `active`, QGA is running, and you are ready to benefit from the associated features. If the output is `inactive`, you may need to install and/or activate QGA.

> ## Opting In

Reviewer 3: ```suggestion
## Opting in
```

Follow these steps to enable QGA on an Instance where it is currently inactive.

### Installation

> Instances created from images older than March 1, 2024, may require manual installation of the `qemu-guest-agent` package:

Reviewer 2: ```suggestion
Instances created from images older than March 1st, 2024, may require manual installation of the `qemu-guest-agent` package:
```

> Instances created from images older than March 1st, 2024, may require manual installation of the `qemu-guest-agent` package:

Reviewer: ```suggestion
Instances created from images older than March 1st, 2024 may require manual installation of the `qemu-guest-agent` package:
```

- **Fedora / AlmaLinux / RockyLinux / CentOS**

  ```bash

>   # yum -y install qemu-guest-agent

Reviewer: ```suggestion
  # yum -y --best install qemu-guest-agent
```

  ```

- **Debian / Ubuntu**

  ```bash
  # apt-get update
  # apt-get -y install qemu-guest-agent
  ```

### Activation

After installing the package, start the `qemu-guest-agent.service` by either:

> - Rebooting your instance, or

Reviewer 2: If we're referring to a Scaleway instance, otherwise ignore my comment.
```suggestion
- Rebooting your Instance, or
```

- Running the following command:

  ```bash
  # systemctl start qemu-guest-agent.service
  ```

## Opting Out

Follow these steps to disable QGA and the associated Scaleway features.

### Deactivation

<Message type="important">

>   Disabling QGA is not recommended, as doing so also disables all [Scaleway features](#what-are-the-features-provided-by-qga) it provides.

Reviewer 2: ```suggestion
  Disabling QGA is not recommended, as doing so also disables all the [Scaleway features](#what-are-the-features-provided-by-qga) it provides.
```

</Message>

To stop and disable QGA, run:

```bash
# systemctl stop qemu-guest-agent.service
# systemctl mask qemu-guest-agent.service
```

This stops the service and prevents it from starting on subsequent reboots.

### Deinstallation (Optional)

<Message type="note">

>   Deinstalling QGA is not strictly necessary to opt out. [Deactivating the service](#deactivation) is sufficient.

Reviewer 2: ```suggestion
  You do not necessarily need to deinstall QGA to opt out. [Deactivating the service](#deactivation) is sufficient.
```
